aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/subscription/RealSubscriptionManager.java

```diff
@@ -205,28 +208,31 @@ public void onError(Exception e) {
                         RealSubscriptionManager.this.removeListener(subObj.subscription, unsubscribeMap.get(subObj));
                         RealSubscriptionManager.this.unsubscribe(subObj.subscription);
                     }
-                    clientConnected.countDown();
+                    allClientsConnectedLatch.countDown();
                 }
             });
-            newClients.add(client);
         }
 
+
         try {
-            clientConnected.await();
+            allClientsConnectedLatch.await();
+            Log.d(TAG, "Made [" + newClients.size() + "] MQTT clients");
 
             synchronized (clients) {
```

Comment:
probably we can remove this synchronized block since the whole method is synchronized

Reply:
Good call. Removed the unnecessary synchronized block.